python/src/robyn/modeling/pareto/pareto_optimizer.py

```diff
@@ -0,0 +1,131 @@
+# pyre-strict
+
+from typing import List, Dict, Any, Optional
+import pandas as pd
+import numpy as np
+from dataclasses import dataclass
+
+from robyn.data.entities.mmmdata import MMMData
+from robyn.modeling.entities.modeloutputs import ModelOutputs
+from robyn.modeling.pareto.response_curve_calculator import ResponseCurveCalculator
+from robyn.modeling.pareto.immediate_carryover_calculator import ImmediateCarryoverCalculator
+from robyn.modeling.pareto.pareto_utils import pareto_front
+
+
+@dataclass
+class ParetoResult:
+    """
+    A dataclass to hold the results of Pareto optimization.
+
+    Attributes:
+        pareto_solutions (List[str]): List of Pareto-optimal solution IDs.
+        pareto_fronts (int): Number of Pareto fronts considered.
+        result_hyp_param (pd.DataFrame): Hyperparameters of Pareto-optimal solutions.
+        x_decomp_agg (pd.DataFrame): Aggregated decomposition results.
+        result_calibration (Optional[pd.DataFrame]): Calibration results, if applicable.
+        media_vec_collect (pd.DataFrame): Collected media vectors.
+        x_decomp_vec_collect (pd.DataFrame): Collected decomposition vectors.
+        plot_data_collect (Dict[str, Any]): Data for various plots.
+        df_caov_pct_all (pd.DataFrame): Carryover percentage data.
+    """
+
+
+class ParetoOptimizer:
+    """
+    A class to perform Pareto optimization on marketing mix models.
+
+    This class orchestrates the Pareto optimization process, including data preparation,
+    Pareto front calculation, response curve calculation, and plot data preparation.
+
+    Attributes:
+        input_collect (MMMData): Input data for the marketing mix model.
```

Comment:
use mmm_data or mmmdata as variable name. Let's be consistent with other parts of the codebase